test/hotspot/jtreg/compiler/loopopts/TestIntCountedLoopLongLimit.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.loopopts;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Asserts;
+
+import java.util.Random;
+
+/**
+ * @test
+ * @bug 8336759
+ * @summary test long limits in int counted loops are speculatively converted to int for counted loop
+ *         optimizations
+ * @library /test/lib /
+ * @requires vm.compiler2.enabled
+ * @run driver compiler.loopopts.TestIntCountedLoopLongLimit
+ */
+public class TestIntCountedLoopLongLimit {
+    private static final Random RNG = jdk.test.lib.Utils.getRandomInstance();
```

Comment:
I suggest to rather use an `import` instead of a fully qualified name:
```suggestion
import jdk.test.lib.Asserts;
import jdk.test.lib.Utils;

import java.util.Random;

/**
 * @test
 * @bug 8336759
 * @summary test long limits in int counted loops are speculatively converted to int for counted loop
 *         optimizations
 * @library /test/lib /
 * @requires vm.compiler2.enabled
 * @run driver compiler.loopopts.TestIntCountedLoopLongLimit
 */
public class TestIntCountedLoopLongLimit {
    private static final Random RNG = Utils.getRandomInstance();
```

Reply:
Oops. I don't know how that happened. Nice catch!